src/test/scala/zio/prelude/ZSetSpec.scala

```diff
@@ -16,7 +16,7 @@ object ZSetSpec extends DefaultRunnableSpec {
     }
 
   def genZSet[R <: Random with Sized, A, B](a: Gen[R, A], b: Gen[R, B]): Gen[R, ZSet[A, B]] =
-    Gen.mapOf(a, b).map(ZSet.fromMap)
+    Gen.mapOf(a, b).map(ZSet.fromMap(_: Map[A, B]))
```

Comment:
The need to use type annotations in unfortunate here. It seems like what this PR buys us is not having to copy the map if it is a `HashMap` versus a normal map, but most of the time that information is not captured in the time signature even if the underlying representation is a `HashMap`. Does it help if we call the methods `fromMap` and `fromHashMap`? Does feel like we are exposing an implementation detail a bit there.